internal/ingress/controller/controller.go

```diff
@@ -167,11 +170,79 @@ func (n *NGINXController) syncIngress(item interface{}) error {
 	if !n.isForceReload() && n.runningConfig.Equal(&pcfg) {
 		glog.V(3).Infof("skipping backend reload (no changes detected)")
 		return nil
+	} else if !n.isForceReload() && len(n.runningConfig.Backends) == len(pcfg.Backends) {
```

Comment:
@zrdaley this is how I ended up implementing it. The idea is to detect _only_ Endpoints changes in case of any other change we fall back to reloading.